### PR TITLE
Refactoring Home page

### PR DESCRIPTION
**Pull request by zpawn**
Story https://app.shortcut.com/deskpro/story/161047/wrike-improvements

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,22 +1,43 @@
...
+
+  const debounceElementEvent = useDebouncedCallback((_, __, payload: EventPayload) => {
+    return match(payload.type)
+      .with("changePage", () => isNavigatePayload(payload) && navigate(payload.path))
+      .run();
+  }, 100);
+
+  useDeskproAppEvents({
+    // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+    // @ts-ignore
```
Comment (HappyPaul55) [inline src/App.tsx:26]: We shouldn't need to ignore types if they're correct. What's the exception here?

File: src/hooks/debounce.ts [hunk 2]
```diff
@@ -1,12 +1,11 @@
...
 
-export default function useDebounce<T>(
-  value: T,
-  delay: number
-): {
+type UseDebounce = <T>(value: T, delay: number) => {
   debouncedValue: T;
   setDebouncedValue: React.Dispatch<React.SetStateAction<T>>;
-} {
+};
```
Comment (HappyPaul55) [inline src/hooks/debounce.ts:6]: Can this file be removed since we have imported a `use-debounce` package instead?

File: vite.config.ts [hunk 3]
```diff
@@ -1,13 +1,21 @@
+import path from "path";
 import { defineConfig } from "vite";
 import react from "@vitejs/plugin-react";
 import copy from "rollup-plugin-copy";
 
+const PORT = process.env.VITE_DEV_SERVER_PORT as undefined|number;
```
Comment (HappyPaul55) [inline vite.config.ts:6]: All environment variables are strings, not numbers. This typecast is inaccurate.

**Review by HappyPaul55 — CHANGES_REQUESTED** [review]

File: src/hooks/debounce.ts (hunk 2 above)
Comment (zpawn) [reply src/hooks/debounce.ts:6]: sure, will be removed after refactoring all pages

File: src/App.tsx (hunk 1 above)
Comment (zpawn) [reply src/App.tsx]: fixed

**Review by HappyPaul55 — APPROVED** [review]

**Pull request merged**